src/test/scala/cromwell/CromwellTestkitSpec.scala

```diff
@@ -134,14 +134,23 @@ with DefaultTimeout with ImplicitSender with WordSpecLike with Matchers with Bef
     TestActorRef(new WorkflowManagerActor(dataAccess, new LocalBackend))
   }
 
-  def runWdl(fsm: TestFSMRef[WorkflowState, WorkflowActor.WorkflowFailure, WorkflowActor], eventFilter: EventFilter, expectedOutputs: Map[FullyQualifiedName, WdlValue] = Map.empty): Unit = {
+  def runWdl(fsm: TestFSMRef[WorkflowState, WorkflowActor.WorkflowFailure, WorkflowActor],
+             eventFilter: EventFilter,
+             expectedOutputs: Map[FullyQualifiedName, WdlValue] = Map.empty,
+             terminalState: WorkflowState = WorkflowSucceeded): Unit = {
     assert(fsm.stateName == WorkflowSubmitted)
     eventFilter.intercept {
       fsm ! WorkflowActor.Start
       within(timeoutDuration) {
         awaitCond(fsm.stateName == WorkflowRunning)
-        awaitCond(fsm.stateName.isTerminal)
-        fsm.stateData should be(WorkflowActor.NoFailureMessage)
+        awaitCond(fsm.stateName == terminalState)
+
+        if (terminalState == WorkflowSucceeded)
+          fsm.stateData should be(WorkflowActor.NoFailureMessage)
+
+        if (terminalState == WorkflowFailed)
+          fsm.stateData.isInstanceOf[WorkflowActor.FailureMessage] shouldEqual true
```

Comment:
Any particular reason why the test for `FailureMessage` needs an `isIntanceOf` but not the one for `NoFailureMessage` ?
`fsm.stateData should be(WorkflowActor.NoFailureMessage)` looks better to me.